File: datagen/src/main/java/juuxel/adorn/datagen/GeneratorConfig.java
package juuxel.adorn.datagen;

import java.util.Map;
import java.util.Set;

public record GeneratorConfig(
    Set<MaterialEntry<WoodMaterial>> woods,
    Set<MaterialEntry<StoneMaterial>> stones,
    Set<MaterialEntry<ColorMaterial>> colors,
    ConditionType conditionType,
    Map<String, String> rootReplacements,
    Overlay overlay,
    String className
) {
    public record MaterialEntry<M extends Material>(
        M material,
        Set<String> exclude,
        Map<String, String> replace
    ) {
    }
}

File: datagen/src/main/kotlin/juuxel/adorn/datagen/GeneratorConfig.kt
package juuxel.adorn.datagen

data class GeneratorConfig(
    val woods: Set<MaterialEntry<WoodMaterial>>,
    val stones: Set<MaterialEntry<StoneMaterial>>,
    val wools: Set<MaterialEntry<WoolMaterial>>,
    val genericMaterials: Set<MaterialEntry<GenericMaterial>>,
    val conditionType: ConditionType,
    val rootReplacements: Map<String, String>,
) {
    data class MaterialEntry<out M : Material>(
        val material: M,
        val exclude: Set<String>,
        val replace: Map<String, String>
    )
}
